### PR TITLE
Dont set context profile on OpenGL < 3.2

### PR DESCRIPTION
**Pull request by Beyley**
# Summary of the PR
On GLFW, it will error out if you set the OpenGL context profile on OpenGL versions less then 3.2

# Related issues, Discord discussions, or proposals
https://discord.com/channels/521092042781229087/587346162802229298/964611543406878780

### CODE REVIEW

File: src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs [hunk 1]
```diff
@@ -326,12 +326,13 @@ protected override void CoreInitialize(WindowOptions opts)
...
-                WindowHintOpenGlProfile.OpenGlProfile,
-                opts.API.Profile == ContextProfile.Core ? OpenGlProfile.Core : OpenGlProfile.Compat
-            );
+            if((opts.API.Version.MajorVersion == 3 && opts.API.Version.MinorVersion >= 2) || opts.API.Version.MajorVersion > 3)
+                // Set API profile
+                _glfw.WindowHint
+                (
+                    WindowHintOpenGlProfile.OpenGlProfile,
+                    opts.API.Profile == ContextProfile.Core ? OpenGlProfile.Core : OpenGlProfile.Compat
+                );
```
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs:335]: Usually lenient on style but this is a small PR, happy to merge after this quick fix is applied ;)

```suggestion
            if ((opts.API.Version.MajorVersion == 3 && opts.API.Version.MinorVersion >= 2) || opts.API.Version.MajorVersion > 3)
            {
                // Set API profile
                _glfw.WindowHint
                (
                    WindowHintOpenGlProfile.OpenGlProfile,
                    opts.API.Profile == ContextProfile.Core ? OpenGlProfile.Core : OpenGlProfile.Compat
                );
            }
```

**Review by HurricanKai — COMMENTED** [review]
Not an expert, but shouldn't we set Any Profile in these cases instead of not providing the hint? The docs say
>  If requesting an OpenGL version below 3.2, GLFW_OPENGL_ANY_PROFILE must be used.

**Review by Perksey — APPROVED** [review]
Beauuuuuuuuuutiful

**Comment by Beyley** [comment]
> Not an expert, but shouldn't we set Any Profile in these cases instead of not providing the hint? The docs say
> 
> > If requesting an OpenGL version below 3.2, GLFW_OPENGL_ANY_PROFILE must be used.

oh i entirely missed that on the docs, but it does still work in my testing

**Comment by Perksey** [comment]
I think it defaults to any profile, happy to do either.

**Pull request merged**